File: core/src/com/spacefist/weapons/Missile.java
/***Doncai******/
package com.spacefist.weapons;

import com.spacefist.GameData;
import com.spacefist.entities.Ship;
import com.spacefist.managers.ProjectileManager;
import com.spacefist.weapons.abst.Weapon;

/**
 * Represents a cluster of three missiles.
 */
public class Missile implements Weapon {
    private ProjectileManager projectileManager;
    private GameData gameData;

    public Missile(GameData gameData) {
        this.gameData = gameData;

        projectileManager = gameData.getProjectileManager();
    }

    /**
     * Fires a missile in the direction the ship is facing.
     */
    @Override
    public void fire() {
        Ship ship = gameData.getShip();

        float shipWidth = ship.getRectangle().getWidth();

        int projectileX = (int) ((ship.getX() + (shipWidth / 2)) - 6);
        int projectileY = (int) ((ship.getY() - (35 * gameData.getScreenScale())));

        projectileManager.fireMissile(projectileX, projectileY);
    }
}
/***Doncai******/

File: core/src/com/spacefist/weapons/Missile.kt
/***Doncai */
package com.spacefist.weapons

import com.spacefist.GameData
import com.spacefist.entities.Ship
import com.spacefist.managers.ProjectileManager
import com.spacefist.weapons.abst.Weapon

/**
 * Represents a cluster of three missiles.
 */
class Missile(private val gameData: GameData) : Weapon {
    private val projectileManager: ProjectileManager

    init {

        projectileManager = gameData.projectileManager
    }

    /**
     * Fires a missile in the direction the ship is facing.
     */
    override fun fire() {
        val ship = gameData.ship

        val shipWidth = ship.rectangle.getWidth()
        val shipHeight = ship.rectangle.getWidth()

        val projectileX = (ship.x + shipWidth / 2 - 6).toInt()
        val projectileY = (ship.y + shipHeight * 1.5f).toInt()

        projectileManager.fireMissile(projectileX, projectileY)
    }
}
/***Doncai */